scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala

```diff
@@ -59,8 +68,17 @@ object DateOps extends java.io.Serializable {
    * Return the guessed format for this datestring
    */
   private[scalding] def getFormatObject(s: String): Option[Format] = {
-    val formats: List[Format] = List(Format.DATE_WITH_DASH, Format.DATEHOUR_WITH_DASH,
-      Format.DATETIME_WITH_DASH, Format.DATETIME_HMS_WITH_DASH, Format.DATETIME_HMSM_WITH_DASH)
+    val formats: List[Format] = List(
+      Format.DATE_WITHOUT_DASH,
```

Comment:
please put `WITH_DASH` first and all the new formats after so existing use cases are not slowed down.